### PR TITLE
 add's get all articles end point and retrieves articles with comment…

### PR DESCRIPTION
**Pull request by SpencerCGriffiths**
… count and no body

### CODE REVIEW

File: app/__tests__/app.test.js [hunk 1]
```diff
@@ -122,6 +81,42 @@ describe("GET /articles/:article_id", () => {
     })
 })
 
+describe("GET /api/articles", () => {
+    describe("-- functionality tests", () => {
+        test("200: responds with 200 status code and all articles, articles should be in descending date order, have a comment count and no body key", () => {
```
Comment (saimakhanom) [inline app/__tests__/app.test.js]: you're testing quite a few different things here, you should split this test into two - one for the articles having the right keys and one to test the sorting

File: app/__tests__/app.test.js [hunk 2]
```diff
@@ -122,6 +81,42 @@ describe("GET /articles/:article_id", () => {
...
+            return request(app)
+            .get("/api/articles")
+            .expect(200)
+            .then(({body}) => {
+                const newestArticle = {
```
Comment (saimakhanom) [inline app/__tests__/app.test.js]: have a look at the jest-sorted library to test whether the articles are sorted in the correct order

File: app/__tests__/app.test.js [hunk 3]
```diff
@@ -122,6 +81,42 @@ describe("GET /articles/:article_id", () => {
...
+                    article_id: 3,
+                    title: 'Eight pug gifs that remind me of mitch',
+                    topic: 'mitch',
+                    author: 'icellusedkars',
+                    created_at: '2020-11-03T09:12:00.000Z',
+                    votes: 0,
+                    article_img_url: 'https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700',
+                    num_of_comments: '2'
```
Comment (saimakhanom) [inline app/__tests__/app.test.js]: this property should be called comment_count

File: app/__tests__/app.test.js [hunk 4]
```diff
@@ -83,36 +83,27 @@ describe("GET /articles/:article_id", () => {
...
-                    author: 'icellusedkars',
-                    created_at: '2020-01-07T14:08:00.000Z',
-                    votes: 0,
-                    article_img_url: 'https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700',
-                    num_of_comments: '0'
-                }
-                expect(body.articles.length).toBe(13)
-                expect(body.articles[0]).toEqual(newestArticle)
-                expect(body.articles[12]).toEqual(oldestArticle)
+                body.articles.forEach((article) => {
```
Comment (aloush-dev) [inline app/__tests__/app.test.js:92]: Don't forget to keep a length check!

**Review by aloush-dev — APPROVED** [review]
Looks good.
Nice job implementing feedback.
Please make sure to add a length check back into your test before merging.

**Pull request merged**